plugins/hls-pragmas-plugin/src/Ide/Plugin/Pragmas.hs

```diff
@@ -192,30 +195,32 @@ allPragmas =
 
 -- ---------------------------------------------------------------------
 flags :: [T.Text]
-flags = map (T.pack . stripLeading '-') $ flagsForCompletion False
+flags = map T.pack $ flagsForCompletion False
 
 completion :: PluginMethodHandler IdeState 'LSP.Method_TextDocumentCompletion
 completion _ide _ complParams = do
     let (LSP.TextDocumentIdentifier uri) = complParams ^. L.textDocument
-        position = complParams ^. L.position
+        cursorPos@(Position l c) = complParams ^. L.position
     contents <- lift $ LSP.getVirtualFile $ toNormalizedUri uri
     fmap LSP.InL $ case (contents, uriToFilePath' uri) of
         (Just cnts, Just _path) ->
-            result <$> VFS.getCompletionPrefix position cnts
+            result <$> VFS.getCompletionPrefix cursorPos cnts
             where
                 result (Just pfix)
                     | "{-# language" `T.isPrefixOf` line
-                    = map buildCompletion
-                        (Fuzzy.simpleFilter (VFS.prefixText pfix) allPragmas)
+                    = map mkLanguagePragmaCompl $
+                        Fuzzy.simpleFilter (VFS.prefixText pfix) allPragmas
                     | "{-# options_ghc" `T.isPrefixOf` line
-                    =  map buildCompletion
-                        (Fuzzy.simpleFilter (VFS.prefixText pfix) flags)
+                    = let flagPrefix = getGhcOptionPrefix cursorPos cnts
```

Comment:
The logic here feels a little bad; we're passing in `pfix` here and no longer using it at all. Maybe we can refactor it a bit?

Reply:
I wanted to do localized change in fear of breaking existing functionality.
But ok, will look into rewriting the whole thing so that it doesn't use VFS.getCompletionPrefix from lsp packae at all. I actually didn't notice there were two versions of that (one in ghcide and one in lsp).

Reply:
Yes, that's what I meant. I didn't meant to say that the pragmas plugin shouldn't be using the generic `getCompletionPrefix` function (although that's almost certainly true, for the same reasons!), just that the particular branch that you're changing now sits oddly.

> I actually didn't notice there were two versions of that (one in ghcide and one in lsp).

I'm going to kill the `lsp` one, so ignore that.

Reply:
On reflection, I don't want to block this on doing the refactoring, so I'm happy to leave it up to you whether you want to do that now or not. If not, maybe leave a note explaining the weirdness.

Reply:
I'll get back to this and try if I can clean it up. It just needs a bit more focus time that I'm unlikely to get during workdays :smile: